src/ModelGeneration.ts

```diff
@@ -238,10 +238,19 @@ function createHandlebarsHelpers(generationOptions: IGenerationOptions): void {
             ? entityName
             : `{${entityName}}`
     );
-    Handlebars.registerHelper("strictMode", () =>
-        generationOptions.strictMode !== "none"
-            ? generationOptions.strictMode
-            : ""
+    Handlebars.registerHelper(
+        "strictMode",
+        (defaultValue?: string | object) => {
+            if (generationOptions.strictMode === "smart") {
+                return defaultValue && typeof defaultValue === "string"
```

Comment:
I had to filter defaultValues with `typeof defaultValue === "string"` because I figured out that object are sent for relations.
Do you think relations should also be optional members ?